src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/converter/AgreementConverter.java

```diff
@@ -6,23 +6,32 @@
 import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import lombok.RequiredArgsConstructor;
 import uk.gov.crowncommercial.dts.scale.service.agreements.model.dto.AgreementDetail;
 import uk.gov.crowncommercial.dts.scale.service.agreements.model.dto.AgreementSummary;
+import uk.gov.crowncommercial.dts.scale.service.agreements.model.dto.DataType;
+import uk.gov.crowncommercial.dts.scale.service.agreements.model.dto.EvaluationType;
 import uk.gov.crowncommercial.dts.scale.service.agreements.model.dto.LotDetail;
 import uk.gov.crowncommercial.dts.scale.service.agreements.model.dto.LotType;
+import uk.gov.crowncommercial.dts.scale.service.agreements.model.dto.RelatedAgreementLot;
 import uk.gov.crowncommercial.dts.scale.service.agreements.model.entity.CommercialAgreement;
 import uk.gov.crowncommercial.dts.scale.service.agreements.model.entity.Lot;
+import uk.gov.crowncommercial.dts.scale.service.agreements.model.entity.LotRelatedLot;
 import uk.gov.crowncommercial.dts.scale.service.agreements.model.entity.Sector;
+import uk.gov.crowncommercial.dts.scale.service.agreements.service.AgreementService;
 
 /**
  * Encapsulates conversion logic between DTOs and Entities.
  */
+@RequiredArgsConstructor
 @Component
 public class AgreementConverter {
 
   @Autowired
   private ModelMapper modelMapper;
 
+  private final AgreementService service;
+
   @PostConstruct
   public void init() {
```

Comment:
SonarLint is starting to complain about the complexity of this method.. I don't see it as an issue though, it's pretty clear what's going on (just one to keep in mind for future extensions)

Reply:
Ah yes - didn't spot that. I can pull the converters out into separate classes - should get rid of that - I'll have a look

Reply:
I've refactored those out now